cypress-io/cypress
# chore: adding snapshots to the log:added/changed

### Additional details

* Added `protocolEnabled` to the Cypress config by returning it from the server config based on whether or not the `protocolManager` exists.
* Whenever a `snapshot` is created, if the protocol has been enabled, we add a snapshot containing the name and timestamp to the `snapshots` array. The snapshots are then passed to the protocol during `log:added/changed`.

### Steps to test

* Checkout `mschile/snapshots` branch from the services repo
* Record a spec
* Verify the snapshots are included in the `log:added/changed` events.

### How has the user experience changed?

<img width="378" alt="Screenshot 2023-05-15 at 9 09 00 AM" src="https://github.com/cypress-io/cypress/assets/2002044/90fc09ad-4515-49b8-8bcd-d7589fb58e7c">

### PR Tasks


- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
